src/ZoneServer/World/Spawner/MonsterSpawner.cs

```diff
@@ -149,15 +144,44 @@ public MonsterSpawner(int monsterClassId, int minAmount, int maxAmount, string m
 		}
 
 		/// <summary>
-		/// Spawns the given number of monsters.
+		/// Adds a spawn point where this spawner can generate mobs into.
+		/// </summary>
+		/// <param name="mapClassName"></param>
+		/// <param name="area"></param>
+		/// <returns></returns>
+		public MonsterSpawnPoint AddSpawnPoint(string mapClassName, IShape area)
+		{
+			if (!ZoneServer.Instance.World.TryGetMap(mapClassName, out var map))
+				throw new ArgumentException($"Map '{mapClassName}' not found.");
+
+			var spawnPoint = new MonsterSpawnPoint(map, area);
+			this.SpawnPoints.Add(spawnPoint);
+
+			return spawnPoint;
+		}
+
+		/// <summary>
+		/// Initializes the population by setting current monster amount to zero
+		/// </summary>
+		public void InitializePopulation()
+		{
+			this.Amount = 0;
+		}
+
+		/// <summary>
+		/// Spawns the given number of monsters in a random spawn point.
 		/// </summary>
 		public void Spawn(int amount)
 		{
+			// Gets random spawn point
+			var index = _rnd.Next(0, this.SpawnPoints.Count);
+			var spawnPoint = this.SpawnPoints[index];
+
 			for (var i = 0; i < amount; ++i)
 			{
-				if (!this.TryGetRandomPosition(out var pos))
+				if (!this.TryGetRandomPosition(spawnPoint, out var pos))
```

Comment:
Would it perhaps make more sense to move the random position getter to the spawn point class?